bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveIndicatorCommandClass.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zwave.internal.protocol.commandclass;
+
+import org.openhab.binding.zwave.internal.config.ZWaveDbCommandClass;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage;
+import org.openhab.binding.zwave.internal.protocol.ZWaveController;
+import org.openhab.binding.zwave.internal.protocol.ZWaveEndpoint;
+import org.openhab.binding.zwave.internal.protocol.ZWaveNode;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage.SerialMessageClass;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage.SerialMessagePriority;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage.SerialMessageType;
+import org.openhab.binding.zwave.internal.protocol.event.ZWaveCommandClassValueEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamOmitField;
+
+/**
+ * Handles the Indicator command class. 
+ * @author Pedro Paixao
+ * @since 1.8.0
+ */
+
+@XStreamAlias("indicatorCommandClass")
+public class ZWaveIndicatorCommandClass extends ZWaveCommandClass implements ZWaveIndicatorCommands {
+
+	@XStreamOmitField
+	private static final Logger logger = LoggerFactory.getLogger(ZWaveIndicatorCommandClass.class);
+	
+	private static final int INDICATOR_SET = 0x01;
+	private static final int INDICATOR_GET = 0x02;
+	private static final int INDICATOR_REPORT = 0x03;
+
+	private boolean isGetSupported = true;
+
+	/**
+	 * Creates a new instance of the ZWaveIndicatorCommandClass class.
+	 * @param node the node this command class belongs to
+	 * @param controller the controller to use
+	 * @param endpoint the endpoint this Command class belongs to
+	 */
+	public ZWaveIndicatorCommandClass(ZWaveNode node,
+			ZWaveController controller, ZWaveEndpoint endpoint) {
+		super(node, controller, endpoint);
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public CommandClass getCommandClass() {
+		return CommandClass.INDICATOR;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void handleApplicationCommandRequest(SerialMessage serialMessage,
+			int offset, int endpoint) {
+		logger.debug("NODE {}: Received Indicator Request", this.getNode().getNodeId());
+		int command = serialMessage.getMessagePayloadByte(offset);
+		switch (command) {
+			case INDICATOR_SET:
+				logger.debug("NODE {}: Indicator Set sent to the controller will be processed as Indicator Report", this.getNode().getNodeId());
+				// Process this as if it was a value report.
+				processIndicatorReport(serialMessage, offset, endpoint);
```

Comment:
Yes, but the binding shouldn't ever receive this message should it? If you SET the device indicator, then you SEND a SET - you don't receive it. If you request the state, then you RECEIVE a REPORT. There are only a couple of instances when you would ever reveice a SET - the BASIC command class does this because the BASIC command class is sent for a lot of different things, but I don't think any other command classes will ever receive the SET command unless something bad happens.

I could be wrong here though - do you actually see this command being received?